Mathlib/Data/Set/Lattice.lean

```diff
@@ -2041,6 +2032,24 @@ theorem iUnion_univ_pi {ι : α → Type*} (t : (a : α) → ι a → Set (π a)
 
 end Pi
 
+section Directed
+
+theorem directedOn_iUnion {r} {f : ι → Set α} (hd : Directed (· ⊆ ·) f)
+    (h : ∀ x, DirectedOn r (f x)) : DirectedOn r (⋃ x, f x) := by
+  simp only [DirectedOn, exists_prop, mem_iUnion, exists_imp]
+  exact fun a₁ b₁ fb₁ a₂ b₂ fb₂ =>
+    let ⟨z, zb₁, zb₂⟩ := hd b₁ b₂
+    let ⟨x, xf, xa₁, xa₂⟩ := h z a₁ (zb₁ fb₁) a₂ (zb₂ fb₂)
+    ⟨x, ⟨z, xf⟩, xa₁, xa₂⟩
+#align set.directed_on_Union Set.directedOn_iUnion
```

Comment:
We want to give any downstream users of this theorem a pointer where it went, like this:

```suggestion
#align set.directed_on_Union Set.directedOn_iUnion

@[deprecated (since = "2024-05-05")]
alias directed_on_iUnion := directedOn_iUnion
```
(Writing on mobile, so I hope I got that right)

Reply:
```suggestion
#align set.directed_on_Union Set.directedOn_iUnion

@[deprecated (since := "2024-05-05")]
alias directed_on_iUnion := directedOn_iUnion
```